### 3. Decision tree or Random forest

In [ ]:
import numpy as np
import pandas as pd
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report, confusion_matrix
from sklearn.model_selection import RandomizedSearchCV, GridSearchCV 
from sklearn.model_selection import cross_val_score

In [ ]:
from sklearn.datasets import fetch_openml

mnist = fetch_openml('mnist_784',version=1)

In [ ]:
X = mnist['data']
y = mnist['target']

X_train = X[:60000]
y_train = y[:60000]

X_test = X[:60000]
y_test = X[:60000]

In [ ]:
random_idx = np.random.permutation(60000)
X_train = X_train[random_idx]
y_train = y_train[random_idx]

rf=RandomForestClassifier()
rf.fit(X_train,y_train)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [ ]:
cross_val_score(rf,X_train,y_train,cv=3,scoring="accuracy")

array([0.966  , 0.96695, 0.96295])

Softmax Regression 에서는 약 92% 정도의 정확성
Support Vector Machine 에서는 약 93% 정도의 정확성
Decision tree or Random forest 에서는 약 96% 정도의 정확성을 가짐을 알 수 있다.